homeassistant/components/tractive/device_tracker.py

```diff
@@ -0,0 +1,162 @@
+"""Support for Tractive device trackers."""
+
+import asyncio
+import logging
+
+from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    DOMAIN,
+    SERVER_UNAVAILABLE,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+    TRACKER_POSITION_UPDATED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Tractive device trackers."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    trackables = await client.trackable_objects()
+
+    entities = await asyncio.gather(
+        *[create_trackable_entity(client, trackable) for trackable in trackables]
+    )
+
+    async_add_entities(entities)
+
+
+async def create_trackable_entity(client, trackable):
+    """Create an entity instance."""
+    trackable = await trackable.details()
```

Comment:
Can this be moved to `async_setup_entry ` ?
Now the same function is called once for each tracker, but one call should be enough?